### PR TITLE
feat: handle seg command server (end of game)

### PR DESCRIPTION
**Pull request by mathieurobert1**
I handle the server command **seg**, the end of the game.

When the GUI receive the command **seg**, it close the window.

More details : #72

### CODE REVIEW

File: gui/src/GUIUpdater/GUIUpdater.cpp [hunk 1]
```diff
@@ -442,7 +442,8 @@ void Gui::GUIUpdater::updateTimeUnitModification(const std::vector<std::string>
 void Gui::GUIUpdater::updateEndOfGame(const std::vector<std::string> &data)
 {
     (void)data;
-    return; // TODO: Implement the end of game
+    _gameData->setIsEndGame(true);
+    std::cout << "The game end" << std::endl;
```
Comment (Thomaltarix) [inline gui/src/GUIUpdater/GUIUpdater.cpp:446]: It might be debug. If not, i think checking the winner is a better idea of printing than this

Comment (Thomaltarix) [reply gui/src/GUIUpdater/GUIUpdater.cpp:446]: As you get the winner team in the `N` parameters in the data, you can print the winning team, and if no team wins, just print `No team wins` or something like this

**Review by Thomaltarix — CHANGES_REQUESTED** [review]
A little change related with the seg implementation

File: gui/src/GUIUpdater/GUIUpdater.cpp (hunk 1 above)
Comment (mathieurobert1) [reply gui/src/GUIUpdater/GUIUpdater.cpp:446]: Oh my bad, I thought seg was a command with no parameters.

**Review by Thomaltarix — DISMISSED** [review]
Clear for me

**Review by Thomaltarix — APPROVED** [review]
Well...

**Review by Njord201 — APPROVED** [review]

**Review by Njord201 — APPROVED** [review]

**Review by Thomaltarix — APPROVED** [review]
Can you let me approve your pr ?

**Pull request merged**